utils/clflags.ml

```diff
@@ -167,10 +167,18 @@ let with_runtime = ref true;;         (* -with-runtime *)
 
 let keep_docs = ref false              (* -keep-docs *)
 let keep_locs = ref true               (* -keep-locs *)
-let unsafe_string =
-  if Config.safe_string then ref false
-  else ref (not Config.default_safe_string)
-                                   (* -safe-string / -unsafe-string *)
+let get_unsafe_string, set_unsafe_string = (* -safe-string / -unsafe-string *)
+  let r =
+    if Config.safe_string then ref false
+    else ref (not Config.default_safe_string)
+  in
+  (fun () -> !r),
+  (fun v ->
+     if Config.safe_string && v then
+       invalid_arg "Internal error: OCaml has been configured with \
+                    -force-safe-string: unsafe-string is not available"
+     else
+       r := v)
```

Comment:
FWIW, I think this would be clearer if the functions were separated based on `Config.safe_string` (it'll be easier to delete it later when the code is totally removed, as well):

```OCaml
let get_unsafe_string, set_unsafe_string = (* -safe-string / -unsafe-string *)
  if Config.safe_string then
    (fun () -> false), (function false -> () | true -> invalid_arg "...")
  else
    let r = ref (not Config.default_safe_string) in
    (fun () -> !r), (fun v -> r := v)
```